storage/store.go

```diff
@@ -1430,20 +1430,72 @@ func (s *Store) WaitForRangeScanCompletion() int64 {
 // updateStoreStatus updates the store's status proto.
 func (s *Store) updateStoreStatus() {
 	now := s.ctx.Clock.Now().WallTime
+	timestamp := proto.Timestamp{WallTime: now}
 	scannerStats := s.scanner.Stats()
 
+	// Get the zone configs.
+	zoneMap, err := s.Gossip().GetInfo(gossip.KeyConfigZone)
+	if err != nil || zoneMap == nil {
+		log.Error("unable to get zone configs")
+		return
+	}
+
+	// Get the leader count and replication count.
+	// TODO(bram): Consider moving this to be part of the range scanner directly.
+	var leaderRangeCount, replicatedRangeCount, availableRangeCount int32
+	s.mu.Lock()
+	for raftID, rng := range s.ranges {
+		zoneConfig := zoneMap.(PrefixConfigMap).MatchByPrefix(rng.Desc().StartKey).Config.(*proto.ZoneConfig)
+		raftStatus := s.RaftStatus(raftID)
+		if raftStatus == nil {
+			continue
+		}
+		if raftStatus.SoftState.RaftState == raft.StateLeader {
+			leaderRangeCount++
+			// TODO(bram): Compare attributes of the stores so we can track
+			// ranges that have enough replicas but still need to be migrated
+			// onto nodes with the desired attributes.
+			if len(raftStatus.Progress) >= len(zoneConfig.ReplicaAttrs) {
+				replicatedRangeCount++
+			}
+
+			// If the range has the leader lease, then it's available.
+			if _, expired := rng.HasLeaderLease(timestamp); !expired {
+				availableRangeCount++
+			} else {
+				// If there is no leader lease, then as long as more than 50%
+				// of the replicas are current then it is available.
+				current := 0
+				for _, progress := range raftStatus.Progress {
```

Comment:
If the raft group hasn't started then the RaftStatus method will return nil and this block will be skipped. The range will not included in either leaderRangeCount or availableRangeCount (so it will effectively look like a follower). 

One possible solution would be to start all the raft groups via a scanner. This makes the whole thing less lazy but we still get the benefit of spreading the initial elections out over the first 10 minutes instead of doing them all simultaneously at startup. If we don't do that and we still want to incorporate raft log positions into our availability metric (and I'm not sure we do), we'll need some monitor process to go out and query all replicas of inactive raft groups for their current log positions.